### DISCUSSION TITLE
"Block domains using filters and hosts files" - what does this option do?

### DISCUSSION THREAD

**Discussion by morpheus1211** [AdguardTeam/AdGuardHome#3337] [Q&A] [answered]
Apologies in advance for what (to me) seems like an extremely dumb question, yet I cannot find the answer here or elsewhere.

I'm running Adguard Home on a Raspberry Pi, installed through DietPi (v0.106.3).

**On the Settings -> General settings tab, there is a checkbox labeled "Block domains using filters and hosts files" but I'm unable to notice any difference in which domains are filtered when this is checked vs. unchecked. So the question is what does this checkbox do?**

This seems to change the filtering_enabled option in AdGuardHome.yaml, but that doesn't seem to have any effect on what AdGuard Home does.

For now, I have only the default Adguard DNS filter list under FIlters -> DNS Blocklists.

If "Block domains using filters and hosts files" is unchecked, I had expected that other DNS features like DNS re-writes would still work, but the filter lists under Filters -> DNS Blocklists would no longer be effective, but this is not what happens. 

I'm testing on a MacOS client and clearing DNS caches before doing look-ups using "sudo dscacheutil -flushcache; sudo killall -HUP mDNSResponder". If for example in a terminal I run "nslookup doubleclick.net" (you can pick any other domain on the default Adguard DNSS filter list), the DNS server responds with 0.0.0.0 regardless of whether the "Block domains using filters and host files" is checked. I realize that I can click "Disable protection" on the Dashboard (which disables both the filter lists and DNS rewrites, and I assume, the allow lists, etc.) and then nslookup will give me the actual public IP of doubleclick.net. So then what does the "Block domains using filters and host files" check box actually do?

Is this likely to be:
(1) a bug, in which case I'm happy file a bug report? (but I doubt that's it)
(2) me misunderstanding how/what that checkbox is support to do?
(3) some kind of mis-configuration in the version of Adguard that DietPi distributes that I'm not realizing?

Everything else on my Adguard configuration is mostly as it came out of the box (except for changing which upstream DNS servers are used)

I guess a secondary question, which was mentioned here is therefore: is there a way to disable the block lists temporarily without disabling DNS re-writes? This seems hinted at in this issue: https://github.com/AdguardTeam/AdGuardHome/issues/1558, which is still marked as open, so I assume the answer to that is no?

Thanks in advance to anyone for their help!

**Comment by ameshkov** [comment, marked as answer]
> On the Settings -> General settings tab, there is a checkbox labeled "Block domains using filters and hosts files" but I'm unable to notice any difference in which domains are filtered when this is checked vs. unchecked. So the question is what does this checkbox do?

If it's disabled, AdGuard Home will not apply the blocklists that are configured, so nothing will be blocked at all.

Also, I cannot reproduce the issue, unchecking this box does unblock `doubleclick.net` for me.

Could it be that you have a "Client" with a custom configuration which overrides the general settings?

**Comment by morpheus1211** [comment]
EDIT: Please ignore this; I'm an idiot. This was an issue in v0.106.3 (the latest "release"), but using the edge channel fixes this (haven't yet checked the beta channel), so it was obviously resolved somewhere along the way.

ORIGINAL POST:
Sorry for the delay (didn't set up my Github notifications correctly). I'm literally using a stock configuration installed using the command on the readme. There's no clients with custom configurations created.

I've also tried it on two different machines, a Mac (Big Sur) and on Linux (Raspberry Pi / DietPi). Both have exactly the same behavior; this setting doesn't seem to have any impact. Checking or unchecking that box does change the "filtering_enabled" item in the AdGuardHome.yaml file (to true or false, as expected), but nothing else seems to change. I've tried restarting AdGuard Home also.

I'm pretty sure I should report this as an "issue"

Steps to reproduce:
1. Clean install of OS (in my case DietPi on a Raspberry Pi; also tried on a Mac using Big Sur with the same results)
2. Run the following to install AdGuard Home:
```
curl -s -S -L https://raw.githubusercontent.com/AdguardTeam/AdGuardHome/master/scripts/install.sh | sh -s -- -v
```
3. Go to the URL indicated (in my case, 192.168.x.x:3000) and set up AdGuard home using all defaults and setting the password.
4. Set client to use Raspberry Pi as DNS server.
5. Flush DNS cache on client.
6. Lookup a blocked domain, and see that it returns 0.0.0.0
```
nslookup doubleclick.net
```
7. Go to the management control panel -> Settings -> General settings, uncheck "Block domains using filters and hosts files"
8. Flush the client's DNS cache.
9. Lookup a blocked domain; it should return the domain's IP address, but doesn't
```
nslookup doubleclick.net
```
(The above still returns 0.0.0.0 for me, again using two different devices.)

Ameshkov; are you using the latest version or do you have any other settings set?
